### PR TITLE
feat: Add ReplaceNamed(), InsertBeforeNamed(), and InsertAfterNamed()

### PR DESCRIPTION
**Pull request by muir**
New functions that allow manipulating the injection chain using the names assigned with `Provide()`

### CODE REVIEW

File: example_reorder_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"fmt"
+
+	"github.com/muir/nject"
+)
+
+// This demonstrates how it to have a default that gets overridden by
+// by later inputs using Reorder
+func ExampleReorder() {
```
Comment (muir) [inline example_reorder_test.go:11]: Moved as-is

File: replace.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+// and the replace directives can be duplicated.
+//
+// When you add a name, with Provide(), you can add it to a collection
+// thus naming multiple providers with the same name.
+//
+// Likewise, when you tag a provider with InsertAfterName, you can
+// be tagging a colleciton, not an individual.
+func (c *Collection) handleReplaceByName() (err error) {
+	defer func() {
+		// if err != nil {
```
Comment (sharnoff) [inline replace.go:54]: is this meant to be commented out?

File: replace.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+			afterLastMove := lastMove.next
+			insertBefore(firstLast.first, firstMove, lastMove)
+			n = afterLastMove.prev
+		case n.fm.insertAfterName != "":
+			name := n.fm.insertAfterName
+			firstLast, err := getTarget(name, "insert after")
+			if err != nil {
+				return err
+			}
+			firstMove, lastMove := snip(n, func(n *node) bool { return n.fm.insertBeforeName == name })
```
Comment (sharnoff) [inline replace.go:238]:
```suggestion
			firstMove, lastMove := snip(n, func(n *node) bool { return n.fm.insertAfterName == name })
```

File: replace.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+			n = afterLastMove.prev
+		default:
+			// nothing
+		}
+	}
+
+	// step 4, convert back to list
+	contents := make([]*provider, 0, len(c.contents))
+	for n := head.next; n != tail; n = n.next {
+		if n.fm != nil && n.fm.fn != nil {
```
Comment (sharnoff) [inline replace.go:250]: can this condition be false?

**Review by sharnoff — APPROVED** [review]

**Pull request merged**